Description: Do not show system users in the "frequent users" list
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=587708
Bug-Ubuntu: https://launchpad.net/bugs/395281

Index: gdm-2.32.0/gui/simple-greeter/gdm-user-manager.c
===================================================================
--- gdm-2.32.0.orig/gui/simple-greeter/gdm-user-manager.c	2010-09-15 20:58:11.000000000 +0200
+++ gdm-2.32.0/gui/simple-greeter/gdm-user-manager.c	2010-12-07 15:10:00.597502000 +0100
@@ -1887,6 +1887,7 @@
         gboolean res;
         char    *username;
         gulong   frequency;
+        struct passwd *pwent;
         GdmUser *user;
 
         frequency = 0;
@@ -1902,6 +1903,18 @@
                 return;
         }
 
+        /* do not show system users; we cannot use gdm_user_manager_get_user()
+         * here since this creates/signals users as a side effect */
+        pwent = getpwnam (username);
+        if (pwent == NULL) {
+                g_warning ("Unable to lookup user name %s: %s", username, g_strerror (errno));
+                return;
+        }
+        if (pwent->pw_uid < DEFAULT_MINIMAL_UID) {
+                g_debug ("GdmUserManager: excluding user '%s'", username);
+                return;
+        }
+
         user = gdm_user_manager_get_user (manager, username);
         if (user == NULL) {
                 g_debug ("GdmUserManager: unable to lookup user '%s'", username);
